networks/suzuka/suzuka-full-node/src/partial.rs

```diff
@@ -177,10 +177,23 @@ where
 
 			// get the transactions
 			let mut block_transactions = Vec::new();
+			let block_metadata = self.executor.build_block_metadata(
+				HashValue::sha3_256_of(block_id.as_bytes()),
+				block_timestamp
+			).await?;
+			let block_metadata_transaction = SignatureVerifiedTransaction::Valid(
+				Transaction::BlockMetadata(
+					block_metadata
+				)
+			);
+			block_transactions.push(block_metadata_transaction);
```

Comment:
Maybe even lower it into maptos-opt-executor, then?

Reply:
Yeah, we may have to reevaluate at some point. But, for now, we are going to assume that the executor can correctly decide the block height based on, well, the number of blocks it has seen. 

This is, at least on the surface, a viable assumption because the if it is an honest node, it should fetch all the blocks up from the DA (and when we have state syncing from other nodes that have done so).

However, if we do need have it come up from the DA, the complexity we will have to deal with is that Celestia blob heights are continually pushed regardless of whether additional blobs have been added to a given namespace.

Reply:
In theory, the sequencer could address by keeping its own counter. But, this could be problematic.